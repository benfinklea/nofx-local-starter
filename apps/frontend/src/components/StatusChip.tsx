import * as React from 'react';
import Chip from '@mui/material/Chip';

type StatusVariant = 'success' | 'error' | 'info' | 'warning' | 'default';

export interface StatusChipProps {
  status?: string | null;
  size?: 'small' | 'medium';
}

function resolveVariant(status?: string | null): StatusVariant {
  if (!status) return 'default';
  const normalized = status.toLowerCase();
  if (['completed', 'succeeded', 'success'].includes(normalized)) return 'success';
  if (['failed', 'cancelled', 'error'].includes(normalized)) return 'error';
  if (['running', 'in_progress', 'queued', 'pending'].includes(normalized)) return 'info';
  if (['warning', 'refused', 'incomplete'].includes(normalized)) return 'warning';
  return 'default';
}

export const StatusChip = React.memo(function StatusChip({ status, size = 'small' }: StatusChipProps) {
  const variant = resolveVariant(status);
  return <Chip label={status ?? 'unknown'} color={variant} size={size} variant={variant === 'default' ? 'outlined' : 'filled'} />;
});

export default StatusChip;
