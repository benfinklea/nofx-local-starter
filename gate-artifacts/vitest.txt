
 RUN  v2.1.9 /Volumes/Development/nofx-local-starter
      Coverage enabled with v8

 % Coverage report from v8
-------------------|---------|----------|---------|---------|-------------------
File               | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-------------------|---------|----------|---------|---------|-------------------
All files          |       0 |     4.08 |    4.08 |       0 |                   
 ...trol-plane/src |       0 |        0 |       0 |       0 |                   
  main.ts          |       0 |        0 |       0 |       0 | 1-12              
 ...ne/src/modules |       0 |        0 |       0 |       0 |                   
  app.module.ts    |       0 |        0 |       0 |       0 | 1-11              
 ...e/src/services |       0 |        0 |       0 |       0 |                   
  db.service.ts    |       0 |        0 |       0 |       0 | 1-18              
  queue.service.ts |       0 |        0 |       0 |       0 | 1-24              
 ...-plane/src/web |       0 |        0 |       0 |       0 |                   
  ...controller.ts |       0 |        0 |       0 |       0 | 1-9               
  ...controller.ts |       0 |        0 |       0 |       0 | 1-50              
 .../worker-ts/src |       0 |        0 |       0 |       0 |                   
  index.ts         |       0 |        0 |       0 |       0 | 1-48              
 ...ges/shared/src |       0 |        0 |       0 |       0 |                   
  types.ts         |       0 |        0 |       0 |       0 |                   
 scripts           |       0 |        0 |       0 |       0 |                   
  createBucket.ts  |       0 |        0 |       0 |       0 | 1-18              
  importModels.ts  |       0 |        0 |       0 |       0 | 1-50              
  runGate.js       |       0 |        0 |       0 |       0 | 1-74              
  ...WriteRules.ts |       0 |        0 |       0 |       0 | 1-29              
 src/api           |       0 |        0 |       0 |       0 |                   
  loader.ts        |       0 |        0 |       0 |       0 | 1-25              
  main.ts          |       0 |        0 |       0 |       0 | 1-149             
  planBuilder.ts   |       0 |        0 |       0 |       0 | 1-56              
 src/api/routes    |       0 |        0 |       0 |       0 |                   
  auth.ts          |       0 |        0 |       0 |       0 | 1-21              
  dev.ts           |       0 |        0 |       0 |       0 | 1-22              
  gates.ts         |       0 |        0 |       0 |       0 | 1-57              
  models.ts        |       0 |        0 |       0 |       0 | 1-119             
  settings.ts      |       0 |        0 |       0 |       0 | 1-34              
  ui.ts            |       0 |        0 |       0 |       0 | 1-38              
 src/lib           |       0 |        0 |       0 |       0 |                   
  auth.ts          |       0 |        0 |       0 |       0 | 1-37              
  db.ts            |       0 |        0 |       0 |       0 | 1-12              
  events.ts        |       0 |        0 |       0 |       0 | 1-7               
  logger.ts        |       0 |        0 |       0 |       0 | 1-2               
  ...lImporters.ts |       0 |        0 |       0 |       0 | 1-95              
  models.ts        |       0 |        0 |       0 |       0 | 1-39              
  queue.ts         |       0 |        0 |       0 |       0 | 1                 
  settings.ts      |       0 |        0 |       0 |       0 | 1-133             
  supabase.ts      |       0 |        0 |       0 |       0 | 1-17              
 src/lib/queue     |       0 |        0 |       0 |       0 |                   
  RedisAdapter.ts  |       0 |        0 |       0 |       0 | 1-28              
  index.ts         |       0 |        0 |       0 |       0 | 1-8               
 src/models        |       0 |        0 |       0 |       0 |                   
  router.ts        |       0 |        0 |       0 |       0 | 1-120             
 ...dels/providers |       0 |        0 |       0 |       0 |                   
  anthropic.ts     |       0 |        0 |       0 |       0 | 1-14              
  gemini.ts        |       0 |        0 |       0 |       0 | 1-16              
  http.ts          |       0 |        0 |       0 |       0 | 1-12              
  openai.ts        |       0 |        0 |       0 |       0 | 1-35              
 src/policy        |       0 |        0 |       0 |       0 |                   
  dbWritePolicy.ts |       0 |        0 |       0 |       0 | 1-9               
 src/shared        |       0 |        0 |       0 |       0 |                   
  types.ts         |       0 |        0 |       0 |       0 | 1-12              
 src/tools         |       0 |        0 |       0 |       0 |                   
  codegen.ts       |       0 |        0 |       0 |       0 | 1-17              
 src/worker        |       0 |        0 |       0 |       0 |                   
  main.ts          |       0 |        0 |       0 |       0 | 1-27              
  runner.ts        |       0 |        0 |       0 |       0 | 1-38              
 ...orker/handlers |       0 |    14.28 |   14.28 |       0 |                   
  codegen.ts       |       0 |        0 |       0 |       0 | 1-59              
  db_write.ts      |       0 |        0 |       0 |       0 | 1-129             
  gate.ts          |       0 |        0 |       0 |       0 | 1-145             
  git_pr.ts        |       0 |        0 |       0 |       0 | 1-141             
  loader.ts        |       0 |        0 |       0 |       0 | 1-16              
  manual.ts        |       0 |        0 |       0 |       0 | 1-63              
  types.ts         |       0 |        0 |       0 |       0 |                   
-------------------|---------|----------|---------|---------|-------------------
No test files found, exiting with code 0


[33mThe CJS build of Vite's Node API is deprecated. See https://vite.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.[39m
include: src/**/*.unit.test.ts
exclude:  **/node_modules/**, **/dist/**, **/cypress/**, **/.{idea,git,cache,output,temp}/**, **/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build,eslint,prettier}.config.*
