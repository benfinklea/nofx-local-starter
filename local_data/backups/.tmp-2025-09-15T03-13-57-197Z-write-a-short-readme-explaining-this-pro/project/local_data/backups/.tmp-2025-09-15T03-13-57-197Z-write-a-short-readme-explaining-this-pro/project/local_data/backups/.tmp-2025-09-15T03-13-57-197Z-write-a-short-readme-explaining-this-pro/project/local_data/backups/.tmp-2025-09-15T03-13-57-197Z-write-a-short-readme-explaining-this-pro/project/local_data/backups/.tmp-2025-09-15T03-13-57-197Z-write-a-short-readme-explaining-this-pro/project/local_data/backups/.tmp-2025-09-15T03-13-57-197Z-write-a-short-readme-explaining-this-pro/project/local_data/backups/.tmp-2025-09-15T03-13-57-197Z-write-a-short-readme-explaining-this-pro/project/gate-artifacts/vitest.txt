
 RUN  v2.1.9 /Volumes/Development/nofx-local-starter
      Coverage enabled with v8

 % Coverage report from v8
-------------------|---------|----------|---------|---------|-------------------
File               | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-------------------|---------|----------|---------|---------|-------------------
All files          |       0 |     3.63 |    3.63 |       0 |                   
 ...trol-plane/src |       0 |        0 |       0 |       0 |                   
  main.ts          |       0 |        0 |       0 |       0 | 1-12              
 ...ne/src/modules |       0 |        0 |       0 |       0 |                   
  app.module.ts    |       0 |        0 |       0 |       0 | 1-11              
 ...e/src/services |       0 |        0 |       0 |       0 |                   
  db.service.ts    |       0 |        0 |       0 |       0 | 1-18              
  queue.service.ts |       0 |        0 |       0 |       0 | 1-24              
 ...-plane/src/web |       0 |        0 |       0 |       0 |                   
  ...controller.ts |       0 |        0 |       0 |       0 | 1-9               
  ...controller.ts |       0 |        0 |       0 |       0 | 1-50              
 .../worker-ts/src |       0 |        0 |       0 |       0 |                   
  index.ts         |       0 |        0 |       0 |       0 | 1-48              
 ...ges/shared/src |       0 |        0 |       0 |       0 |                   
  types.ts         |       0 |        0 |       0 |       0 |                   
 scripts           |       0 |        0 |       0 |       0 |                   
  createBucket.ts  |       0 |        0 |       0 |       0 | 1-18              
  importModels.ts  |       0 |        0 |       0 |       0 | 1-50              
  runGate.js       |       0 |        0 |       0 |       0 | 1-74              
  ...WriteRules.ts |       0 |        0 |       0 |       0 | 1-29              
  ...erredOrder.ts |       0 |        0 |       0 |       0 | 1-16              
 src               |       0 |        0 |       0 |       0 |                   
  simple.ts        |       0 |        0 |       0 |       0 | 1-17              
 src/api           |       0 |        0 |       0 |       0 |                   
  loader.ts        |       0 |        0 |       0 |       0 | 1-25              
  main.ts          |       0 |        0 |       0 |       0 | 1-147             
  planBuilder.ts   |       0 |        0 |       0 |       0 | 1-56              
 src/api/routes    |       0 |        0 |       0 |       0 |                   
  auth.ts          |       0 |        0 |       0 |       0 | 1-21              
  dev.ts           |       0 |        0 |       0 |       0 | 1-48              
  gates.ts         |       0 |        0 |       0 |       0 | 1-65              
  models.ts        |       0 |        0 |       0 |       0 | 1-119             
  queue.ts         |       0 |        0 |       0 |       0 | 1-42              
  settings.ts      |       0 |        0 |       0 |       0 | 1-37              
  ui.ts            |       0 |        0 |       0 |       0 | 1-49              
 src/lib           |       0 |        0 |       0 |       0 |                   
  artifacts.ts     |       0 |        0 |       0 |       0 | 1-24              
  auth.ts          |       0 |        0 |       0 |       0 | 1-37              
  db.ts            |       0 |        0 |       0 |       0 | 1-12              
  events.ts        |       0 |        0 |       0 |       0 | 1-5               
  logger.ts        |       0 |        0 |       0 |       0 | 1-2               
  ...lImporters.ts |       0 |        0 |       0 |       0 | 1-95              
  models.ts        |       0 |        0 |       0 |       0 | 1-39              
  queue.ts         |       0 |        0 |       0 |       0 | 1                 
  settings.ts      |       0 |        0 |       0 |       0 | 1-133             
  store.ts         |       0 |        0 |       0 |       0 | 1-206             
  supabase.ts      |       0 |        0 |       0 |       0 | 1-19              
 src/lib/queue     |       0 |        0 |       0 |       0 |                   
  MemoryAdapter.ts |       0 |        0 |       0 |       0 | 1-73              
  RedisAdapter.ts  |       0 |        0 |       0 |       0 | 1-45              
  index.ts         |       0 |        0 |       0 |       0 | 1-11              
 src/models        |       0 |        0 |       0 |       0 |                   
  router.ts        |       0 |        0 |       0 |       0 | 1-120             
 ...dels/providers |       0 |        0 |       0 |       0 |                   
  anthropic.ts     |       0 |        0 |       0 |       0 | 1-14              
  gemini.ts        |       0 |        0 |       0 |       0 | 1-16              
  http.ts          |       0 |        0 |       0 |       0 | 1-12              
  openai.ts        |       0 |        0 |       0 |       0 | 1-35              
 src/policy        |       0 |        0 |       0 |       0 |                   
  dbWritePolicy.ts |       0 |        0 |       0 |       0 | 1-9               
 src/shared        |       0 |        0 |       0 |       0 |                   
  types.ts         |       0 |        0 |       0 |       0 | 1-12              
 src/tools         |       0 |        0 |       0 |       0 |                   
  codegen.ts       |       0 |        0 |       0 |       0 | 1-17              
 src/worker        |       0 |        0 |       0 |       0 |                   
  main.ts          |       0 |        0 |       0 |       0 | 1-40              
  runner.ts        |       0 |        0 |       0 |       0 | 1-36              
 ...orker/handlers |       0 |    14.28 |   14.28 |       0 |                   
  codegen.ts       |       0 |        0 |       0 |       0 | 1-52              
  db_write.ts      |       0 |        0 |       0 |       0 | 1-129             
  gate.ts          |       0 |        0 |       0 |       0 | 1-94              
  git_pr.ts        |       0 |        0 |       0 |       0 | 1-141             
  loader.ts        |       0 |        0 |       0 |       0 | 1-16              
  manual.ts        |       0 |        0 |       0 |       0 | 1-47              
  types.ts         |       0 |        0 |       0 |       0 |                   
-------------------|---------|----------|---------|---------|-------------------
No test files found, exiting with code 0


[33mThe CJS build of Vite's Node API is deprecated. See https://vite.dev/guide/troubleshooting.html#vite-cjs-node-api-deprecated for more details.[39m
include: src/**/*.unit.test.ts
exclude:  **/node_modules/**, **/dist/**, **/cypress/**, **/.{idea,git,cache,output,temp}/**, **/{karma,rollup,webpack,vite,vitest,jest,ava,babel,nyc,cypress,tsup,build,eslint,prettier}.config.*
