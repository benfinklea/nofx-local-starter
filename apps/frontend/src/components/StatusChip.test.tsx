/**
 * StatusChip Component Tests
 * Testing all status variants and edge cases
 */

import { describe, test, expect } from 'vitest';
import { render, screen } from '@testing-library/react';
import { StatusChip } from './StatusChip';

describe('StatusChip Component', () => {
  describe('Status Variant Resolution', () => {
    test('renders success variant for completed status', () => {
      const { container } = render(<StatusChip status="completed" />);
      const chip = container.querySelector('.MuiChip-filled');
      expect(chip).toBeInTheDocument();
      expect(screen.getByText('completed')).toBeInTheDocument();
    });

    test('renders success variant for succeeded status', () => {
      const { container } = render(<StatusChip status="succeeded" />);
      expect(screen.getByText('succeeded')).toBeInTheDocument();
    });

    test('renders success variant for success status', () => {
      const { container } = render(<StatusChip status="success" />);
      expect(screen.getByText('success')).toBeInTheDocument();
    });

    test('renders error variant for failed status', () => {
      const { container } = render(<StatusChip status="failed" />);
      const chip = container.querySelector('.MuiChip-filled');
      expect(chip).toBeInTheDocument();
      expect(screen.getByText('failed')).toBeInTheDocument();
    });

    test('renders error variant for cancelled status', () => {
      render(<StatusChip status="cancelled" />);
      expect(screen.getByText('cancelled')).toBeInTheDocument();
    });

    test('renders error variant for error status', () => {
      render(<StatusChip status="error" />);
      expect(screen.getByText('error')).toBeInTheDocument();
    });

    test('renders info variant for running status', () => {
      const { container } = render(<StatusChip status="running" />);
      const chip = container.querySelector('.MuiChip-filled');
      expect(chip).toBeInTheDocument();
      expect(screen.getByText('running')).toBeInTheDocument();
    });

    test('renders info variant for in_progress status', () => {
      render(<StatusChip status="in_progress" />);
      expect(screen.getByText('in_progress')).toBeInTheDocument();
    });

    test('renders info variant for queued status', () => {
      render(<StatusChip status="queued" />);
      expect(screen.getByText('queued')).toBeInTheDocument();
    });

    test('renders info variant for pending status', () => {
      render(<StatusChip status="pending" />);
      expect(screen.getByText('pending')).toBeInTheDocument();
    });

    test('renders warning variant for warning status', () => {
      const { container } = render(<StatusChip status="warning" />);
      const chip = container.querySelector('.MuiChip-filled');
      expect(chip).toBeInTheDocument();
      expect(screen.getByText('warning')).toBeInTheDocument();
    });

    test('renders warning variant for refused status', () => {
      render(<StatusChip status="refused" />);
      expect(screen.getByText('refused')).toBeInTheDocument();
    });

    test('renders warning variant for incomplete status', () => {
      render(<StatusChip status="incomplete" />);
      expect(screen.getByText('incomplete')).toBeInTheDocument();
    });

    test('renders default variant for unknown status', () => {
      const { container } = render(<StatusChip status="unknown_status" />);
      const chip = container.querySelector('.MuiChip-outlined');
      expect(chip).toBeInTheDocument();
      expect(screen.getByText('unknown_status')).toBeInTheDocument();
    });

    test('renders default variant when status is null', () => {
      const { container } = render(<StatusChip status={null} />);
      const chip = container.querySelector('.MuiChip-outlined');
      expect(chip).toBeInTheDocument();
      expect(screen.getByText('unknown')).toBeInTheDocument();
    });

    test('renders default variant when status is undefined', () => {
      const { container } = render(<StatusChip />);
      const chip = container.querySelector('.MuiChip-outlined');
      expect(chip).toBeInTheDocument();
      expect(screen.getByText('unknown')).toBeInTheDocument();
    });
  });

  describe('Case Insensitivity', () => {
    test('handles uppercase status correctly', () => {
      render(<StatusChip status="COMPLETED" />);
      expect(screen.getByText('COMPLETED')).toBeInTheDocument();
    });

    test('handles mixed case status correctly', () => {
      render(<StatusChip status="InProgress" />);
      expect(screen.getByText('InProgress')).toBeInTheDocument();
    });

    test('handles lowercase status correctly', () => {
      render(<StatusChip status="failed" />);
      expect(screen.getByText('failed')).toBeInTheDocument();
    });
  });

  describe('Size Prop', () => {
    test('renders small size by default', () => {
      const { container } = render(<StatusChip status="completed" />);
      const chip = container.querySelector('.MuiChip-sizeSmall');
      expect(chip).toBeInTheDocument();
    });

    test('renders medium size when specified', () => {
      const { container } = render(<StatusChip status="completed" size="medium" />);
      const chip = container.querySelector('.MuiChip-sizeMedium');
      expect(chip).toBeInTheDocument();
    });

    test('renders small size when explicitly specified', () => {
      const { container } = render(<StatusChip status="completed" size="small" />);
      const chip = container.querySelector('.MuiChip-sizeSmall');
      expect(chip).toBeInTheDocument();
    });
  });

  describe('Edge Cases', () => {
    test('handles empty string status', () => {
      const { container } = render(<StatusChip status="" />);
      const chip = container.querySelector('.MuiChip-outlined');
      expect(chip).toBeInTheDocument();
      // Empty string renders as empty label, not 'unknown'
      const label = container.querySelector('.MuiChip-label');
      expect(label).toBeInTheDocument();
    });

    test('handles whitespace-only status', () => {
      const { container } = render(<StatusChip status="   " />);
      const label = container.querySelector('.MuiChip-label');
      expect(label).toBeInTheDocument();
      expect(label?.textContent).toBe('   ');
    });

    test('handles status with special characters', () => {
      render(<StatusChip status="status-with-dashes" />);
      expect(screen.getByText('status-with-dashes')).toBeInTheDocument();
    });

    test('handles very long status text', () => {
      const longStatus = 'this_is_a_very_long_status_name_that_might_cause_layout_issues';
      render(<StatusChip status={longStatus} />);
      expect(screen.getByText(longStatus)).toBeInTheDocument();
    });
  });

  describe('Component Memoization', () => {
    test('component is memoized with React.memo', () => {
      expect(StatusChip.$$typeof).toBeDefined();
    });

    test('re-renders with different status', () => {
      const { rerender } = render(<StatusChip status="pending" />);
      expect(screen.getByText('pending')).toBeInTheDocument();

      rerender(<StatusChip status="completed" />);
      expect(screen.getByText('completed')).toBeInTheDocument();
      expect(screen.queryByText('pending')).not.toBeInTheDocument();
    });

    test('re-renders with different size', () => {
      const { rerender, container } = render(<StatusChip status="completed" size="small" />);
      expect(container.querySelector('.MuiChip-sizeSmall')).toBeInTheDocument();

      rerender(<StatusChip status="completed" size="medium" />);
      expect(container.querySelector('.MuiChip-sizeMedium')).toBeInTheDocument();
    });
  });

  describe('Visual Regression', () => {
    test('snapshot for all success statuses', () => {
      const statuses = ['completed', 'succeeded', 'success'];
      statuses.forEach(status => {
        const { container } = render(<StatusChip status={status} />);
        expect(container.firstChild).toMatchSnapshot(`status-chip-${status}`);
      });
    });

    test('snapshot for all error statuses', () => {
      const statuses = ['failed', 'cancelled', 'error'];
      statuses.forEach(status => {
        const { container } = render(<StatusChip status={status} />);
        expect(container.firstChild).toMatchSnapshot(`status-chip-${status}`);
      });
    });

    test('snapshot for all info statuses', () => {
      const statuses = ['running', 'in_progress', 'queued', 'pending'];
      statuses.forEach(status => {
        const { container } = render(<StatusChip status={status} />);
        expect(container.firstChild).toMatchSnapshot(`status-chip-${status}`);
      });
    });

    test('snapshot for all warning statuses', () => {
      const statuses = ['warning', 'refused', 'incomplete'];
      statuses.forEach(status => {
        const { container } = render(<StatusChip status={status} />);
        expect(container.firstChild).toMatchSnapshot(`status-chip-${status}`);
      });
    });
  });

  describe('Accessibility', () => {
    test('chip has proper role', () => {
      render(<StatusChip status="completed" />);
      const chip = screen.getByText('completed').closest('.MuiChip-root');
      expect(chip).toBeInTheDocument();
    });

    test('status text is readable', () => {
      render(<StatusChip status="running" />);
      expect(screen.getByText('running')).toBeVisible();
    });

    test('filled variants are distinguishable from outlined', () => {
      const { container: filledContainer } = render(<StatusChip status="completed" />);
      const { container: outlinedContainer } = render(<StatusChip status="unknown" />);

      expect(filledContainer.querySelector('.MuiChip-filled')).toBeInTheDocument();
      expect(outlinedContainer.querySelector('.MuiChip-outlined')).toBeInTheDocument();
    });
  });
});
