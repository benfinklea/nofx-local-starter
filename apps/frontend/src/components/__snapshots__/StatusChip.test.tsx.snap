// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StatusChip Component > Visual Regression > snapshot for all error statuses > status-chip-cancelled 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorError MuiChip-filledError css-1ihdc1n-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    cancelled
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all error statuses > status-chip-error 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorError MuiChip-filledError css-1ihdc1n-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    error
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all error statuses > status-chip-failed 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorError MuiChip-filledError css-1ihdc1n-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    failed
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all info statuses > status-chip-in_progress 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorInfo MuiChip-filledInfo css-16lfvse-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    in_progress
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all info statuses > status-chip-pending 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorInfo MuiChip-filledInfo css-16lfvse-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    pending
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all info statuses > status-chip-queued 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorInfo MuiChip-filledInfo css-16lfvse-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    queued
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all info statuses > status-chip-running 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorInfo MuiChip-filledInfo css-16lfvse-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    running
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all success statuses > status-chip-completed 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-1fiq0pv-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    completed
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all success statuses > status-chip-succeeded 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-1fiq0pv-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    succeeded
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all success statuses > status-chip-success 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorSuccess MuiChip-filledSuccess css-1fiq0pv-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    success
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all warning statuses > status-chip-incomplete 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorWarning MuiChip-filledWarning css-se6ndb-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    incomplete
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all warning statuses > status-chip-refused 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorWarning MuiChip-filledWarning css-se6ndb-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    refused
  </span>
</div>
`;

exports[`StatusChip Component > Visual Regression > snapshot for all warning statuses > status-chip-warning 1`] = `
<div
  class="MuiChip-root MuiChip-filled MuiChip-sizeSmall MuiChip-colorWarning MuiChip-filledWarning css-se6ndb-MuiChip-root"
>
  <span
    class="MuiChip-label MuiChip-labelSmall css-eccknh-MuiChip-label"
  >
    warning
  </span>
</div>
`;
